### PR TITLE
(DOCSP-18534): Document ThreadSafe property wrapper

### PR DESCRIPTION
**Pull request by cbush**
### Jira

- https://jira.mongodb.org/browse/DOCSP-18534

### Staged Changes (Requires MongoDB Corp SSO)

- https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/threadsafe-wrapper/sdk/ios/advanced-guides/threading/#pass-instances-across-threads

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/ios/README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ This project uses CocoaPods to manage dependencies. To get started, run:
 pod install --repo-update
 ```
 
+
```
Comment (nspektor) [inline examples/ios/README.md:20]: Was this added by accident or do we need the newline?

File: examples/ios/Examples/Threading.swift [hunk 2]
```diff
@@ -163,6 +163,72 @@ class Threading: XCTestCase {
...
+        // @ThreadSafe vars are always optional. If the referenced object is deleted,
+        // the @ThreadSafe var will be nullified.
+        print("Person's name: \(personRef?.name ?? "unknown")")
+
+        // Pass the reference to a background thread
+        DispatchQueue(label: "background").async {
+            autoreleasepool {
+                let realm = try! Realm()
+                try! realm.write {
+                    // Resolve within the transaction to ensure you get the latest changes from other threads.
```
Comment (nspektor) [inline examples/ios/Examples/Threading.swift:188]: This line length is causing the code block on the page to have a horizontal scroll, maybe we could split it into two shorter lines? (cutoff is right after "latest")

File: source/sdk/ios/advanced-guides/threading.txt [hunk 3]
```diff
@@ -131,8 +131,45 @@ Instances of ``Realm``, ``Results``, ``List``, and managed ``Objects``
...
+
+.. versionadded:: 10.17.0
+
 You can pass thread-confined instances to another thread as follows:
 
+1. Use the ``@ThreadSafe`` property wrapper to declare a variable that references the original object. By definition, ``@ThreadSafe``-wrapped variables are always optional.
+#. Pass the ``@ThreadSafe``-wrapped variable to the other thread.
+#. Use the ``@ThreadSafe``-wrapped variable as you would any optional. If the referenced object is removed from the {+realm+}, the referencing variable becomes nil.
+
+.. literalinclude:: /examples/generated/code/start/Threading.codeblock.threadsafe-wrapper.swift
```
Comment (nspektor) [inline source/sdk/ios/advanced-guides/threading.txt:147]: Why is this code block not in an example directive while the other one is?

File: source/sdk/ios/advanced-guides/threading.txt [hunk 4]
```diff
@@ -131,8 +131,45 @@ Instances of ``Realm``, ``Results``, ``List``, and managed ``Objects``
...
+again on that thread. But this can get complicated when the object does
+not have a :ref:`primary key
+<ios-find-a-specific-object-by-primary-key>`. You can use the
+``@ThreadSafe`` wrapper on any object, regardless of whether it has a
+primary key. You can also use it on lists and results, on a struct or
+class property, or on a function parameter.
+
+.. example::
+
+   The following example shows how to use ``@ThreadSafe`` on a function
```
Comment (nspektor) [inline source/sdk/ios/advanced-guides/threading.txt:160]: Maybe add a sentence saying why `@ThreadSafe` makes sense in this example

File: source/sdk/ios/advanced-guides/threading.txt [hunk 5]
```diff
@@ -131,8 +131,45 @@ Instances of ``Realm``, ``Results``, ``List``, and managed ``Objects``
...
+   parameter.
+
+   .. literalinclude:: /examples/generated/code/start/Threading.codeblock.threadsafe-wrapper-function-parameter.swift
+      :language: swift
+
+
+Use the @ThreadSafe Wrapper (Legacy Swift / Objective-C)
```
Comment (nspektor) [inline source/sdk/ios/advanced-guides/threading.txt:167]: Is it still considered a `Wrapper` in the legacy version?

**Review by nspektor — COMMENTED** [review]
Looks great, I left a few questions/comments 😄

File: source/sdk/ios/advanced-guides/threading.txt (hunk 3 above)
Comment (cbush) [reply source/sdk/ios/advanced-guides/threading.txt:147]: This one already has explanatory text. I felt the next one came out of nowhere without a preamble.

**Pull request merged**